src/gui/guiFormSpecMenu.cpp

```diff
@@ -766,108 +766,75 @@ void GUIFormSpecMenu::parseScrollBarOptions(parserData* data, const std::string
-
-	if ((parts.size() == 3) ||
-		((parts.size() > 3) && (m_formspec_version > FORMSPEC_API_VERSION)))
-	{
-		std::vector<std::string> v_pos = split(parts[0],',');
-		std::vector<std::string> v_geom = split(parts[1],',');
-		std::string name = unescape_string(parts[2]);
+	std::vector<std::string> parts = split(element, ';');
 
-		MY_CHECKPOS("image", 0);
-		MY_CHECKGEOM("image", 1);
+	if (parts.size() < 3 || parts.size() > 4 ||
+			(parts.size() > 4 && m_formspec_version <= FORMSPEC_API_VERSION)) {
+		errorstream << "Invalid image element(" << parts.size() << "): '" << element
+			<< "'" << std::endl;
+		return;
+	}
 
-		v2s32 pos;
-		v2s32 geom;
+	std::vector<std::string> v_pos = split(parts[0],',');
+	std::vector<std::string> v_geom = split(parts[1],',');
+	std::string name = unescape_string(parts[2]);
 
-		if (data->real_coordinates) {
-			pos = getRealCoordinateBasePos(v_pos);
-			geom = getRealCoordinateGeometry(v_geom);
-		} else {
-			pos = getElementBasePos(&v_pos);
-			geom.X = stof(v_geom[0]) * (float)imgsize.X;
-			geom.Y = stof(v_geom[1]) * (float)imgsize.Y;
-		}
+	MY_CHECKPOS("image", 0);
+	MY_CHECKGEOM("image", 1);
 
-		if (!data->explicit_size)
-			warningstream<<"invalid use of image without a size[] element"<<std::endl;
+	v2s32 pos;
+	v2s32 geom;
 
-		video::ITexture *texture = m_tsrc->getTexture(name);
-		if (!texture) {
-			errorstream << "GUIFormSpecMenu::parseImage() Unable to load texture:"
-					<< std::endl << "\t" << name << std::endl;
-			return;
-		}
+	if (data->real_coordinates) {
+		pos = getRealCoordinateBasePos(v_pos);
+		geom = getRealCoordinateGeometry(v_geom);
+	} else {
+		pos = getElementBasePos(&v_pos);
+		geom.X = stof(v_geom[0]) * (float)imgsize.X;
+		geom.Y = stof(v_geom[1]) * (float)imgsize.Y;
+	}
 
-		FieldSpec spec(
-			name,
-			L"",
-			L"",
-			258 + m_fields.size(),
-			1
-		);
-		core::rect<s32> rect(pos, pos + geom);
-		gui::IGUIImage *e = Environment->addImage(rect, data->current_parent,
-				spec.fid, 0, true);
-		e->setImage(texture);
-		e->setScaleImage(true);
-		auto style = getDefaultStyleForElement("image", spec.fname);
-		e->setNotClipped(style.getBool(StyleSpec::NOCLIP, m_formspec_version < 3));
-		m_fields.push_back(spec);
+	if (!data->explicit_size)
+		warningstream << "Invalid use of animated_image without a size[] element"
+				<< std::endl;
 
-		// images should let events through
-		e->grab();
-		m_clickthrough_elements.push_back(e);
-		return;
-	}
+	FieldSpec spec(
+		name,
+		L"",
+		L"",
+		258 + m_fields.size(),
+		1
+	);
 
-	if (parts.size() == 2) {
-		std::vector<std::string> v_pos = split(parts[0],',');
-		std::string name = unescape_string(parts[1]);
+	core::rect<s32> rect = core::rect<s32>(pos, pos + geom);
 
-		MY_CHECKPOS("image", 0);
+	core::rect<s32> middle;
+	if (parts.size() >= 4)
+		parseRect(parts[3], &middle);
 
-		v2s32 pos = getElementBasePos(&v_pos);
+	GUIAnimatedImage *e = new GUIAnimatedImage(Environment, data->current_parent,
```

Comment:
Wouldn't it be better not to add the image if the texture is null?

Reply:
Honestly, this could apply to a lot of elements, e.g. empty text on a `label` or `vertlabel`.  However, I'd personally leave these elements in; for instance, `model` can have a `bgcolor` set, and if the element wasn't created if the model provided didn't exist, the bgcolor wouldn't appear.  `image` and `animated_image` currently don't have such properties, but they could quite easily in the future.